dotnetv3/EC2/Tests/EC2WrapperTests.cs

```diff
@@ -0,0 +1,389 @@
+        Assert.NotNull(_associationId);
+    }
+
+    /// <summary>
+    /// Test the ability to stop a running EC2 instance.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact()]
+    [Order(14)]
+    public async Task StopInstancesTest()
+    {
+        try
+        {
+            await _ec2Wrapper.StopInstances(_ec2InstanceId);
+            var hasStopped = false;
+            do
+            {
+                hasStopped = await _ec2Wrapper.WaitForInstanceState(_ec2InstanceId, InstanceStateName.Stopped);
+            } while (!hasStopped);
+
+            Assert.True(hasStopped);
+        }
+        catch (Exception ex)
+        {
+            Assert.True(false, $"Could not stop the instance. Error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Test the ability to start an existing EC2 instance.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact()]
+    [Order(15)]
+    public async Task StartInstancesTest()
+    {
+        try
+        {
+            await _ec2Wrapper.StartInstances(_ec2InstanceId);
+
+            // Wait for the instance state to be running.
+            var isRunning = false;
+            do
+            {
+                isRunning = await _ec2Wrapper.WaitForInstanceState(_ec2InstanceId, InstanceStateName.Running);
+            } while (!isRunning);
+
+            Assert.True(isRunning);
+        }
+        catch (Exception ex)
+        {
+            Assert.True(false, $"Could not start {_ec2InstanceId}. Error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Test the ability to disassociate an IP address from an Amazon EC2
+    /// Elastic IP address.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact()]
+    [Order(16)]
+    public async Task DisassociateIpTest()
+    {
+        var success = await _ec2Wrapper.DisassociateIp(_associationId);
+        Assert.True(success, "Could not disassociate IP address.");
+    }
+
+    /// <summary>
+    /// Test the ability to release an Amazon EC2 Elastic IP address.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact()]
+    [Order(17)]
+    public async Task ReleaseAddressTest()
+    {
+        var success = await _ec2Wrapper.ReleaseAddress(_allocationId);
+        Assert.True(success, "Could not releases the address.");
+    }
+
+    /// <summary>
+    /// Test the ability to terminate an existing EC2 instance.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact()]
+    [Order(18)]
+    public async Task TerminateInstanceTest()
+    {
+        var stateChange = await _ec2Wrapper.TerminateInstances(_ec2InstanceId);
+
+        // Wait for the instance state to be terminated.
+        var hasTerminated = false;
+        do
+        {
+            hasTerminated = await _ec2Wrapper.WaitForInstanceState(_ec2InstanceId, InstanceStateName.Terminated);
+        } while (!hasTerminated);
+
+        Assert.True(hasTerminated);
+    }
+    /// <summary>
```

Comment:
Need a leading line break here.

Reply:
Strangely that wasn't picked up by the linter either. As an experiment, I used the format document command in Visual Studio and that doesn't pick it up either.